CMakeLists.txt

```diff
@@ -4,350 +4,109 @@ if(COMMAND CMAKE_POLICY)
-      )
-
-    link_directories(${PROJECT_BINARY_DIR}/src)
-
-    if (NOT DEFINED BUILD_TESTING OR BUILD_TESTING)
-      set(BUILD_SDF_TEST TRUE)
-    else()
-      set(BUILD_SDF_TEST FALSE)
-    endif()
-
-    if (BUILD_SDF_TEST)
-      add_subdirectory(test)
+# Find ign-cmake
+find_package(ignition-cmake2 2.10 REQUIRED)
+set(IGN_CMAKE_VER ${ignition-cmake2_VERSION_MAJOR})
+
+if (BUILD_SDF)
+  ign_configure_project(
+    NO_IGNITION_PREFIX
+    REPLACE_IGNITION_INCLUDE_PATH sdf)
+
+  message (STATUS "\n\n====== Finding 3rd Party Packages ======")
+  message(STATUS "Building for arch: ${ARCH}")
+
+  #################################################
+  # Find tinyxml2.
+  ign_find_package(TINYXML2 REQUIRED)
+
+  ################################################
+  # Find urdfdom parser. Logic:
+  #
+  #  1. if USE_INTERNAL_URDF is unset, try to use system installation, fallback to internal copy
+  #  2. if USE_INTERNAL_URDF is set to True, use the internal copy
+  #  3. if USE_INTERNAL_URDF is set to False, force to search system installation, fail on error
+  if (NOT DEFINED USE_INTERNAL_URDF OR NOT USE_INTERNAL_URDF)
+    ign_find_package(IgnURDFDOM VERSION 1.0 QUIET)
+    if (NOT IgnURDFDOM_FOUND)
+      if (NOT DEFINED USE_INTERNAL_URDF)
+        # fallback to internal urdf
+        set(USE_INTERNAL_URDF ON)
+      else()
+        ign_build_error("Couldn't find the urdfdom >= 1.0 system installation")
+      endif()
     endif()
-    add_subdirectory(src)
-    add_subdirectory(include/sdf)
-    add_subdirectory(sdf)
-    add_subdirectory(conf)
-    add_subdirectory(doc)
-  endif(BUILD_SDF)
-
-  ########################################
-  # Setup Codecheck
-  include (IgnCodeCheck)
-  set(CPPCHECK_DIRS
-    ${PROJECT_SOURCE_DIR}/src
-    ${PROJECT_SOURCE_DIR}/include
-    ${PROJECT_SOURCE_DIR}/test/integration
-    ${PROJECT_SOURCE_DIR}/test/performance)
-
-  set(CPPCHECK_INCLUDE_DIRS
-    ${PROJECT_BINARY_DIR}
-    ${PROJECT_SOURCE_DIR}/include
-    ${PROJECT_SOURCE_DIR}/test/integration
-    ${PROJECT_SOURCE_DIR}/test/performance)
+  endif()
 
-  # Ignore vendored directories.
-  file(WRITE ${PROJECT_BINARY_DIR}/cppcheck.suppress
-    "*:${PROJECT_SOURCE_DIR}/src/urdf/*\n"
-    )
-  ign_setup_target_for_codecheck()
+  if (USE_INTERNAL_URDF)
+    message(STATUS "Using internal URDF")
+  endif()
 
-  ########################################
-  # Make the package config file
-  configure_file(${CMAKE_SOURCE_DIR}/cmake/sdformat_pc.in
-    ${CMAKE_CURRENT_BINARY_DIR}/cmake/${PROJECT_NAME_LOWER}.pc @ONLY)
-  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/cmake/${PROJECT_NAME_LOWER}.pc DESTINATION
-    ${LIB_INSTALL_DIR}/pkgconfig COMPONENT pkgconfig)
+  #################################################
+  # Find ign command line utility:
+  ign_find_package(ignition-tools)
+
+  ################################################
+  # Find the Python interpreter for running the
+  # check_test_ran.py script
+  find_package(PythonInterp 3 QUIET)
```

Comment:
We need to maintain this deprecated CMake command to support BIonic 3.10.2 version of CMake but should be migrated if new versions don't need to support it.

Reply:
actually, we don't need this at all since ign-cmake is providing `check_test_ran.py`

I'll remove it

Reply:
perfect.

Reply:
removed in https://github.com/ignitionrobotics/sdformat/commit/795e10192d0a03859fab5fbe5c719fa2abd932ba